### PR TITLE
MGDAPI-1004 Update AddOn Delete Watch

### PR DESCRIPTION
**Pull request by CathalOConnorRH**
# Description
Add the ability for hive to remove a cluster using a labeled configmap. 

To Verify:
1. Install RHOAM/RHMI from this branch
2. Once install is complete, add a config map in the integreatly operator namespace called `managed-service-api` / `rhmi` 
3. Label the config map with `api.openshift.com/addon-managed-api-service-delete=true`/ `api.openshift.com/addon-rhmi-operator-delete=true`
4. once the configmap is read the rhmi cr should have a deletion timestamp and the uninstall process should start. 

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

**Comment by Boomatang** [comment]
I have tried this with both fully installed version of RHMI & RHOAM and while both version were still in the install process. Each time it worked as expected.
/lgtm

File: pkg/controller/namespacelabel/namespacelabel_controller.go [hunk 1]
```diff
@@ -130,6 +143,15 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// Watch the SKU rate limits config map
```
Comment (philbrookes) [inline pkg/controller/namespacelabel/namespacelabel_controller.go:146]: Why is this watch on the SKU configmap added?

Comment (CathalOConnorRH) [reply pkg/controller/namespacelabel/namespacelabel_controller.go]: That was a copy paste mistake on the comment. I've update it now.

**Pull request merged**